src/main/flight/alt_hold.c

```diff
@@ -22,118 +22,50 @@
-    // iTerm isn't very important if hover throttle is set carefully and sag compensation is used
-
-    // The altitude D lowpass filter is very important.
-    // The only way to get enough D is to filter the oscillations out.
-    // More D filtering is needed with Baro than with GPS, since GPS is smoother and slower.
-
-    // A major problem is the lag time for motors to arrest pre-existing drops or climbs,
-    // compounded by the lag time from filtering.
-    // If the quad is dropping fast, the motors have to be high for a long time to arrest the drop
-    // this is very difficult for a 'simple' PID controller;
-    // if the PIDs are high enough to arrest a fast drop, they will oscillate under normal conditions
-    // Hence we:
-    // - Enhance D when the absolute velocity is high, ie when we need to strongly oppose a fast drop,
-    //   even though it may cause throttle oscillations while dropping quickly - the average D is what we need
-    // - Prevent excessive iTerm growth when error is impossibly large for iTerm to resolve
-
-    const pidCoefficient_t *pidCoefs =  getAltitudePidCoeffs();
-
-    const float altitudeErrorCm = altHoldState.targetAltitudeCm - getAltitudeCm();
-
-    // P
-    const float pOut = pidCoefs->Kp * altitudeErrorCm;
-
-    // I
-    // input limit iTerm so that it doesn't grow fast with large errors
-    // very important at the start if there are massive initial errors to prevent iTerm windup
-
-    // much less iTerm change for errors greater than 2m, otherwise it winds up badly
-    const float itermNormalRange = 200.0f; // 2m
-    const float itermRelax = (fabsf(altitudeErrorCm) < itermNormalRange) ? 1.0f : 0.1f;
-    pidIntegral += altitudeErrorCm * pidCoefs->Ki * itermRelax * taskIntervalSeconds;
-    // arbitrary limit on iTerm, same as for gps_rescue, +/-20% of full throttle range
-    // ** might not be needed with input limiting **
-    pidIntegral = constrainf(pidIntegral, -200.0f, 200.0f); 
-    const float iOut = pidIntegral;
-
-    // D
     // boost D by 'increasing apparent velocity' when vertical velocity exceeds 5 m/s ( D of 75 on defaults)
     // usually we don't see fast ascend/descend rates if the altitude hold starts under stable conditions
     // this is important primarily to arrest pre-existing fast drops or climbs at the start;
-
     float vel = getAltitudeDerivative(); // cm/s altitude derivative is always available
     const float kinkPoint = 500.0f; // velocity at which D should start to increase
     const float kinkPointAdjustment = kinkPoint * 2.0f; // Precompute constant
     const float sign = (vel > 0) ? 1.0f : -1.0f;
     if (fabsf(vel) > kinkPoint) {
         vel = vel * 3.0f - sign * kinkPointAdjustment;
     }
+    const float verticalVelocity = vel;
 
-    const float dOut = pidCoefs->Kd * vel;
+    const float targetAltitudeCm = altHoldState.targetAltitudeCm;
+    const float targetChangeRate = altHoldState.targetAltitudeAdjustRate;
```

Comment:
```suggestion
```

Reply:
Yes, simpler, wasn't sure why I put them into the floats. Thanks.